src/main/java/com/google/finapp/DatabaseModule.java

```diff
@@ -28,21 +28,25 @@
 final class DatabaseModule extends AbstractModule {
 
   @Override
-  protected void configure() {
-    bind(SpannerDaoInterface.class).to(SpannerDaoImpl.class);
-  }
+  protected void configure() {}
 
   @Provides
   @Singleton
-  DatabaseClient provideDatabaseClient(
+  SpannerDaoInterface provideSpannerDao(
       @ArgsModule.SpannerHost String spannerHost,
       @ArgsModule.SpannerPort int spannerPort,
       @ArgsModule.SpannerProjectId String spannerProjectId,
       @ArgsModule.SpannerInstanceId String spannerInstanceId,
-      @ArgsModule.SpannerDatabaseId String spannerDatabaseId) {
+      @ArgsModule.SpannerDatabaseId String spannerDatabaseId,
+      @ArgsModule.SpannerUseJdbc boolean spannerUseJdbc) {
     SpannerOptions spannerOptions = SpannerOptions.getDefaultInstance();
     Spanner spanner = spannerOptions.toBuilder().build().getService();
-    return spanner.getDatabaseClient(
+    DatabaseClient client = spanner.getDatabaseClient(
         DatabaseId.of(spannerProjectId, spannerInstanceId, spannerDatabaseId));
+    if (spannerUseJdbc) {
+      return new SpannerDaoJDBCImpl(spannerProjectId, spannerInstanceId, spannerDatabaseId);
+    } else {
+      return new SpannerDaoImpl(client);
+    }
```

Comment:
If `spannerUseJdbc`, then we don't need to initialize `spannerOptions`, `spanner`, or `client`. Maybe move those all into the `else` clause (to both make it clear that that is part of the other implementation and to not waste any computation).